openfisca_core/simulations.py

```diff
@@ -50,15 +55,17 @@ def __init__(
         self.create_shortcuts()
 
         # To keep track of the values (formulas and periods) being calculated to detect circular definitions.
-        # See use in formulas.py.
-        # The data structure of requested_periods_by_variable_name is: {variable_name: [period1, period2]}
-        self.requested_periods_by_variable_name = {}
-        self.max_nb_cycles = None
+        # The data structure of computation_stack is:
+        # [('variable_name', 'period1'), ('variable_name', 'period2')]
+        self.computation_stack = []
+        self.invalidated_caches = set()
 
         self.debug = False
         self.trace = False
         self.opt_out_cache = False
 
+        # controls the spirals detection; check for performance impact if > 1
+        self.max_spiral_loops = 1
```

Comment:
Should this really be a Simulation instance attribute? Are we considering allowing users to tune their `max_spiral_loops` ?

Reply:
Yes but it's not a hard decision yet. What would you recommend?
Note: It's not documented in the official documentation.

Reply:
Mmm not sure. But we can keep it as an instance attribute to leave the option later.